webgui/scripts/codecompass/view/component/Text.js

```diff
@@ -483,23 +474,23 @@ function (declare, domClass, dom, style, query, topic, ContentPane, Dialog,
       //   and end column respectively.
       // - line (optional): The line number to jump to when the file is loaded.
       topic.subscribe('codecompass/openFile', function (message) {
-        that.loadFile(message.fileId);
-
         if (!message.selection)
           message.selection = [1, 1, 1, 1];
 
-        that.set('selection', message.selection);
-
-        that.jumpToPos(
-          message.line || 0,
-          message.selection ? message.selection[1] : 0);
-
         urlHandler.setStateValue({
           fid    : message.fileId,
           select : message.selection.join('|'),
           center : that.id
         });
 
+        that.loadFile(message.fileId);
+
+        that.set('selection', message.selection);
+
+        that.jumpToPos(
+          message.line || 0,
+          message.selection ? message.selection[1] : 0);
+
```

Comment:
No, these changes happend because of caching.
- First we set the URL state which will cache file information and other stuff (if related URL information changed ).
- In `loadFile()` we will use the file information in the cache  if the file id is the same, otherwise we will query the file information.